### PR TITLE
Statistics DB: prune old records

### PR DESCRIPTION
**Pull request by janpriessnitz**
Delete old records to keep the statistics DB small.
Maximum age of records to keep can be configured by
CVMFS_STATS_DB_DAYS_TO_KEEP and is 365 by default.

### CODE REVIEW

**Comment by jblomer** [comment]
Thanks!  Please have a look if the `vacuum` sql command is called, which compacts the database after delete.

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest nodestroy

File: cvmfs/statistics_database.cc [hunk 1]
```diff
@@ -490,10 +523,14 @@ bool StatisticsDatabase::UploadStatistics(
...
 
-std::string StatisticsDatabase::GetDBPath(const std::string &repo_name) {
+void StatisticsDatabase::GetDBParams(const std::string &repo_name,
+                                     std::string *path,
+                                     uint32_t *days_to_keep)
+{
   // default location
   const std::string db_default_path =
       "/var/spool/cvmfs/" + repo_name + "/stats.db";
+  const uint32_t default_days_to_keep = 365;
```
Comment (jblomer) [inline cvmfs/statistics_database.cc]: Let's make this a private static const in the header of the `StatisticsDatabase` class

**Pull request merged**